pkg/testing/tools/tools.go

```diff
@@ -81,14 +81,19 @@ func InstallAgentWithPolicy(t *testing.T, ctx context.Context, installOpts atest
 		agentFixture.SetUninstallToken(uninstallToken)
 	}
 
-	err = InstallAgentForPolicy(t, installOpts, agentFixture, kibClient, policy.ID)
+	err = InstallAgentForPolicy(t, ctx, installOpts, agentFixture, kibClient, policy.ID)
 	return policy, err
 }
 
-// InstallAgentForPolicy enrolls the given agent
-// fixture in Fleet using the default Fleet Server, waits for the agent to be
-// online, and returns error or nil.
-func InstallAgentForPolicy(t *testing.T, installOpts atesting.InstallOpts, agentFixture *atesting.Fixture, kibClient *kibana.Client, policyID string) error {
+// InstallAgentForPolicy enrolls the given agent fixture in Fleet using the
+// default Fleet Server, waits for the agent to be online, and returns error or nil.
+// If ctx has a deadline, it'll wait the agent to become online until 500ms
```

Comment:
Why 500ms? What problem is this solving?

Reply:
just giving some slack so there is still some time after this function gives up waiting and the the context to be done.
I can remove if you want

Reply:
Yeah I am not sure the point of it, should the default timeout of 5 minutes be enough, is 500ms really helping?